### DISCUSSION TITLE
How to use @actions in models properly?

### DISCUSSION THREAD

**Discussion by diganta162004** [Nozbe/WatermelonDB#1079] [Q&A]
WatermelonDB with ReactNative 0.64

I am trying to add a generate action inside the model like this:

```
class Events extends Model {
  static table = 'events';

  @field('event_at') event_at;
  @field('trigger') trigger;
  // other fields

  @action async generateRandomData() {
    console.log('ASYNC EVENT ACTION');
    const randomData = [];
    for (let i = 0; i < 10; i++) {
      randomData.push(
          // code to generate data
        ),
      );
    }
    return await this.batch(randomData);
  }
}

export default Events;
```

But how will be able to call the from inside a component function?

Things i have tried so far:
```
database.action(async () => {
      await events.generateRandomData();
 });
```

```
database.collections.get('events').generateRandomData()
```

and some other combinations... but in all places i was getting generateRandomData not defined. 

I tried to find some help in the documentation and examples but couldn't find anything.

**Comment by radex** [comment]
> but in all places i was getting generateRandomData not defined

that's because `generateRandomData not defined`. You're confusing types. `generateRandomData` is a method on `Event`, but `database.collections.get('events')` isn't an Event, it's a Collection.

**Reply by diganta162004** [reply]
So how should i call a Model action from outside? Should  i call it on with an event entry?

Am i even using Actions correctly for this use-case to generate random data? or should i just move it to an external function with batch.

**Reply by rohit3345ksChicmic** [reply]
Hi @radex , can you please post a correct implementation syntax for the same? There aren't many examples on this and neither we do see hints for our methods in editor.
